examples/util/RSocket.h

```diff
@@ -0,0 +1,36 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+
+#pragma once
+
+#include <folly/ExceptionWrapper.h>
+#include <folly/futures/Future.h>
+#include "examples/util/TcpConnectionFactory.h"
+#include "src/Payload.h"
+#include "src/ReactiveStreamsCompat.h"
+#include "src/StandardReactiveSocket.h"
+
+using namespace ::reactivesocket;
+
+/**
+ * Simplified API for client/server
+ */
+namespace rsocket {
+
+class RSocket {
+ public:
+  static std::unique_ptr<RSocket> createClientFactory(
+      std::unique_ptr<ConnectionFactory> connection);
+  RSocket(std::unique_ptr<ConnectionFactory> connection);
+  ~RSocket();
+  /*
+   * Connect asynchronously and return a Future
```

Comment:
Good point regarding fast start. I wonder if the API should be explicit about opting into that, such as connectFast() which communicates that fast start will be done, which means all requests may fail, rather than waiting for a successful connection?